source/engine/graphics/cubemaps/CubeMapStorage.cpp

```diff
@@ -0,0 +1,80 @@
+#include "CubeMapStorage.h"
+#include "graphics/Base.h"
+#include "graphics/GpuAllocator.h"
+#include "graphics/cubemaps/CubeMapManager.h"
+#include "graphics/vk/Initializers.h"
+
+namespace nc::graphics
+{
+CubeMapStorage::CubeMapStorage(Base* base, GpuAllocator* allocator, const nc::GpuAccessorSignals& gpuAccessorSignals)
+    : m_base{base},
+      m_allocator{allocator},
+      m_cubeMaps{},
+      m_sampler{graphics::CreateTextureSampler(base->GetDevice(), vk::SamplerAddressMode::eRepeat)},
+      m_onCubeMapUpdate{gpuAccessorSignals.onCubeMapUpdate->Connect(this, &CubeMapStorage::UpdateBuffer)}
+{
+}
+
+CubeMapStorage::~CubeMapStorage() noexcept
+{
+    for(auto& cubeMap : m_cubeMaps)
```

Comment:
Same as the other destructor - is this necessary? Our destructors should be such that this sort of cleanup is automatic.